packages/sdk/src/db_connection_builder.ts

```diff
@@ -70,6 +71,13 @@ export class DBConnectionBuilder<DBConnection> {
     return this;
   }
 
+  withCompression(
+    compression: 'gzip' | 'none'
+  ): DBConnectionBuilder<DBConnection> {
+    this.#compression = compression;
+    return this;
+  }
```

Comment:
I question whether we actually want to expose `none` as an option to users. We believe that `gzip` (and `brotli`, not used here) provides a better compute-time / networking tradeoff than `none`, which means both:

- We expect `none` to be a worse choice for the user most of the time.
- More importantly, we expect `none` to be a worse choice for the host all of the time.

So why should we allow malicious or stupid clients to request `none` compression?

I made this same argument in the Rust client SDK. There, there's value to having the method but making it unstable, as we can use it internally for testing and experimentation to compare Brotli and Gzip, with none as a sort of baseline, but I don't see any such value here. So I'd propose we simply remove this method from the TypeScript SDK entirely, until such time as we have multiple different useful options which a user might actually want to choose between.

Anyways, if we are going to expose this method, it needs a doc comment.